Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp

```diff
@@ -61,6 +62,14 @@ void UPSOverlayWidget::SetOverlayVisibility(ESlateVisibility VisibilitySlate, bo
 void UPSOverlayWidget::NativeTick(const FGeometry& MyGeometry, float InDeltaTime)
 {
 	Super::NativeTick(MyGeometry, InDeltaTime);
+	if (PSCOverlay)
```

Comment:
Please simplify the logic in Native Tick to the few lines:
- don't use PCSOverlay at all
- don't change visibility at all (its responsibility of other functions)

in tick, only perform next logic:
if (GetVisibility() != Visible)
{
return;
}

Reply:
done with https://github.com/h4rdmol/ProgressionSystem/pull/34/commits/7de39fb2669235d271becbcfe70dd0d18ffad97a